abce/agent.py

```diff
@@ -156,21 +156,36 @@ def __init__(self, id, agent_parameters, simulation_parameters, group, trade_log
         for resource, units, product in resource_endowment:
             self._register_resource(resource, units, product)
 
-    def init(self, simulation_parameters, agent_parameters):
+    def init(self):
         """ This method is called when the agents are build.
         It can be overwritten by the user, to initialize the agents.
-        parameters and agent_parameters are the parameters given in
+        Parameters are the parameters given to
         :py:meth:`abce.Simulation.build_agents`.
 
-        This function has two arguments simulation_parameters and
-        agent_parameters. The two arguments are taken from
-        :py:meth:`abce.Simulation.build_agents`.
+        Example::
 
-        Example:
+            class Student(abce.Agent):
+                def init(self, rounds, age, lazy, school_size):
+                    self.rounds = rounds
+                    self.age = age
+                    self.lazy = lazy
+                    self.school_size = school_size
+
+                def say(self):
+                    print('I am', self.age ' years old and go to a school
+                    that is ', self.school_size')
+
+
+            def main():
+                sim = Simulation()
+                students = sim.build_agents(Student, 'student',
+                                            agent_parameters=[{'age': 12, lazy: True},
+                                                              {'age': 12, lazy: True},
+                                                              {'age': 13, lazy: False},
+                                                              {'age': 14, lazy: True}],
+                                            rounds=50,
+                                            school_size=990)
 
```

Comment:
The rationale for this API change is that in the previous version Agent.init(self, simulation_parameter, agent_parameter), even when no parameters where specified. Further, the modeler had to remember that init is called with two parameters no matter what and he would have to remember the order. 
When I presented abcEconomics in two weeks ago this was flagged as confusing. The new api, provides the parameters specified with the name they were specified in build_agents. Nothing to remember. 
The new api leads also to less boilerplate as we don't need to reference dictionaries in Agent.init anymore.
The new api makes it easy to read agents from an excel sheet or csv file::
 
    import csv  
with open('chess_set.csv') as csvfile:  
    reader = csv.DictReader(csvfile)
agents = sim.build_agents(Agent, 'agent', world_size=(8, 8), agent_parameter=reader)

Reply:
> and he would have to remember the order

Both `agent_parameter` and `parameter` are named arguments with default values, so ordering is not necessary, but helpful for consistent code.

Reply:
> > and he would have to remember the order
> Both agent_parameter and parameter are named arguments with default values, so ordering is not necessary, but helpful for consistent code.

Ah this is not the case; right, the order is necessary.